lib/service/kubernetes.go

```diff
@@ -149,6 +150,19 @@ func (process *TeleportProcess) initKubernetesService(log *logrus.Entry, conn *C
 		log.Info("Started reverse tunnel client.")
 	}
 
+	var dynLabels *labels.Dynamic
+	if len(cfg.Kube.DynamicLabels) != 0 {
+		dynLabels, err = labels.NewDynamic(process.ExitContext(), &labels.DynamicConfig{
+			Labels: cfg.Kube.DynamicLabels,
+			Log:    log,
+		})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		dynLabels.Sync()
+		go dynLabels.Start()
```

Comment:
How does it behave if `initKubernetesService` fails to complete - will it be left running? Can it prevent shutdown on error?